pkg/kev/converter/kubernetes/project_service.go

```diff
@@ -32,42 +32,26 @@ import (
 	"k8s.io/apimachinery/pkg/util/intstr"
 )
 
-// enabled returns Bool telling Kev whether app component is enabled/disabled
-func (p *ProjectService) enabled() bool {
-	if val, ok := p.Labels[config.LabelComponentEnabled]; ok {
-		if v, err := strconv.ParseBool(val); err == nil {
-			return v
-		}
-
-		log.WarnfWithFields(log.Fields{
-			"project-service": p.Name,
-			"enabled":         val,
-		}, "Unable to extract Bool value from %s label. Component will remain enabled.",
-			config.LabelComponentEnabled)
+func NewProjectService(svc composego.ServiceConfig) (ProjectService, error) {
+	cfg, err := config.K8SCfgFromCompose(&svc)
+	if err != nil {
+		return ProjectService{}, err
 	}
 
-	return true
+	return ProjectService{
+		ServiceConfig: svc,
+		K8SConfig:     cfg,
+	}, nil
+}
+
+// enabled returns Bool telling Kev whether app component is enabled/disabled
+func (p *ProjectService) enabled() bool {
+	return !p.K8SConfig.Disabled
 }
 
 // replicas returns number of replicas for given project service
 func (p *ProjectService) replicas() int32 {
-	if val, ok := p.Labels[config.LabelWorkloadReplicas]; ok {
-		replicas, err := strconv.Atoi(val)
-		if err != nil {
-			log.WarnfWithFields(log.Fields{
-				"project-service": p.Name,
-				"replicas":        val,
-			}, "Unable to extract integer value from %s label. Defaulting to 1 replica.",
-				config.LabelWorkloadReplicas)
-
-			return config.DefaultReplicaNumber
-		}
-		return int32(replicas)
-	} else if p.Deploy != nil && p.Deploy.Replicas != nil {
-		return int32(*p.Deploy.Replicas)
-	}
-
-	return config.DefaultReplicaNumber
+	return int32(p.K8SConfig.Workload.Replicas)
```

Comment:
Yay! 🥳

Reply:
That's great, but should probably guard against values passed as strings too? As it stands it'll fail to extrac replicas from the following config:
```yaml
x-k8s:
  workload:
    replicas: "10"
```
Would be good to either support that also, OR handle the error if value is not as expected.

Reply:
This will not pass validation because it won't be the right type. At this point we're operating on top of K8SConfiguration which is a struct with defined types. So passing replicas: "10" would result in either a parsing error or a validation error. In either case it is detected before it gets to project_service